cpp/include/kvikio/file_handle.hpp

```diff
@@ -549,44 +556,51 @@ class FileHandle {
    * This is an asynchronous version of `.write()`, which will be executed in sequence
    * for the specified stream.
    *
+   * When running CUDA v12.1 or older, this function falls back to use `.read()` after
+   * `stream` has been synchronized.
+   *
    * The arguments have the same meaning as in `.write()` but some of them are deferred. That is,
    * the values of `size`, `file_offset` and `devPtr_offset` will not be evaluated until execution
```

Comment:
```suggestion
   * the values pointed to by `size_p`, `file_offset_p` and `devPtr_offset_p` will not be evaluated until execution
```
As for `read`, this is only true if we're not in fallback mode, right?

Reply:
True, in fallback mode we synchronize immediately but I guess it is a implementation detail?

Reply:
Perhaps then I don't understand what it means for an argument to be "deferred" in the cufile API.

How I interpret what is going on is is something like:

```
void *devPtr_base = nullptr;
...
handle = read_async(devPtr_base, ...); // this call doesn't touch the pointer for now.
...
// some time later
*devPtr_base = some_valid_pointer;
// We are in control of when the read is actually submitted, so we can ensure that the
// pointer is set up correctly.
handle.submit(); 
```

But it seems like what you're saying, things are more like:
```
void *devPtr_base = nullptr;
...
handle = read_async(devPtr_base, ...); // this call doesn't (necessarily) touch the pointer immediately
...
// this might not have worked, because the work might already have run?
*devPtr_base = some_valid_pointer;
handle.sync();
```

Certainly the fallback "stream-sync + read" mode in cuda < 12.2 behaves more like the latter: we must arrange for all the pointers to be valid before calling `read_async` in fallback mode, because there's no chance for us to update them later.

Or do I have the wrong end of the stick?

Reply:
I think the goal of the deferred arguments in cuFile is the ability to feed the output of one async cuda function into the next async function without sync.
```cpp
ssize_t *bytes_read = malloc(sizeof(ssize_t));
read_async(..., bytes_read, stream);
cudaMemCpyAsync(..., *bytes_read, stream);
write_async(..., size_p=bytes_read, stream);
```

Reply:
Ah OK, so it's not "set up a plan and fill in the details later". In which case, I withdraw my worries.